### DISCUSSION TITLE
Unable to select query when special characters are present

### DISCUSSION THREAD

**Discussion by tommyboylab** [vercel/next.js#13002] [Help]
# Bug report

While using router.query, I'm unable to select query values which are contained within square brackets

## Describe the bug
While receiving a response from an OAuth provider, I receive a query string with an access token, token type and expiry. While the first parameter is named normally (access_token) the others are encapsulated in square brackets ([expires_in]) because of this, I'm unable to select them.

## To Reproduce
- Query string is different depending on supplier

```
Code: router.query = {
access_token: Some string
raw[access_token]: Some String
raw[token_type]: bearer
raw[expires_in]: Some Number
}
```

router.query.access_token = Works
router.query.raw[expires_in] = Does not

## Expected behaviour
I should be able to select query parameters which are wrapped in square brackets, same as any other parameters.

If needed, I should be able to supply a string to router.query.`raw[access_token]` as to be able to select such params
## Screenshots
NA

## System information

- OS: [macOS]
- Version of Next.js: [9.3.6]

## Additional context
If there is a way to do this already, please let me know.

**Comment by zlwaterfield** [comment]
This is not a Next.js issue, your Javascript syntax is just incorrect. 

Trying using this to access the keys:
```
router.query['raw[access_token]']
```
